jt-threshold/src/main/java/it/geosolutions/jaiext/threshold/ThresholdOpImage.java

```diff
@@ -0,0 +1,1372 @@
+                break;
+            case DataBuffer.TYPE_FLOAT:
+                floatLoop(src, dst, roiTile, roiContainsTile);
+                break;
+            case DataBuffer.TYPE_DOUBLE:
+                doubleLoop(src, dst, roiTile, roiContainsTile);
+                break;
+            default:
+                throw new RuntimeException("Wrong image data type");
+            }
+            dst.copyDataToRaster();
+        } else {
+            // Setting all as NoData
+            int numBands = getSampleModel().getNumBands();
+            double[] background = new double[numBands];
+            switch (destType) {
+            case DataBuffer.TYPE_BYTE:
+                Arrays.fill(background, destNoDataByte);
+                break;
+            case DataBuffer.TYPE_USHORT:
+            case DataBuffer.TYPE_SHORT:
+                Arrays.fill(background, destNoDataShort);
+                break;
+            case DataBuffer.TYPE_INT:
+                Arrays.fill(background, destNoDataInt);
+                break;
+            case DataBuffer.TYPE_FLOAT:
+                Arrays.fill(background, destNoDataFloat);
+                break;
+            case DataBuffer.TYPE_DOUBLE:
+                Arrays.fill(background, destNoDataDouble);
+                break;
+            default:
+                throw new RuntimeException("Wrong image data type");
+            }
+            ImageUtil.fillBackground(dest, destRect, background);
+        }
+    }
+
+    private void byteLoop(RasterAccessor src, RasterAccessor dst, ROI roiTile,
+            boolean roiContainsTile) {
+
+        int srcLineStride = src.getScanlineStride();
+        int srcPixelStride = src.getPixelStride();
+        int[] srcBandOffsets = src.getBandOffsets();
+        byte[][] srcData = src.getByteDataArrays();
+
+        int dstWidth = dst.getWidth();
+        int dstHeight = dst.getHeight();
+        int dstBands = dst.getNumBands();
+        int dstLineStride = dst.getScanlineStride();
+        int dstPixelStride = dst.getPixelStride();
+        int[] dstBandOffsets = dst.getBandOffsets();
+        byte[][] dstData = dst.getByteDataArrays();
+
+        int srcLineOffset = 0;
+        int dstLineOffset = 0;
+
+        int x0 = 0;
+        int y0 = 0;
+
+        int srcX = src.getX();
+        int srcY = src.getY();
+
+        if (caseA || (caseB && roiContainsTile)) {
+            for (int h = 0; h < dstHeight; h++) {
+                int srcPixelOffset = srcLineOffset;
+                int dstPixelOffset = dstLineOffset;
+
+                for (int w = 0; w < dstWidth; w++) {
+
+                    for (int b = 0; b < dstBands; b++) {
+                        double lo = low[b];
+                        double hi = high[b];
+                        double co = constant[b];
+
+                        byte flo = (byte) lo;
+                        byte fhi = (byte) hi;
+                        byte fco = (byte) co;
+
```

Comment:
note that these 3 casts are made W*H times.  Wondering if they can be computed once per band and re-used as optimization (Just an idea, not a requirement)